be/src/storage/binlog_file_writer.cpp

```diff
@@ -483,4 +532,19 @@ void BinlogFileWriter::_reset_pending_context() {
     _pending_page_context->page_content.Clear();
 }
 
+StatusOr<std::shared_ptr<BinlogFileWriter>> BinlogFileWriter::reopen(int64_t file_id, const std::string& file_path,
```

Comment:
Maybe this `BinlogFileWriter` should be `unique_ptr`? Since only it could not be accessed by multiple callers?